include/matx/transforms/cub.h

```diff
@@ -492,87 +494,269 @@ class matxCubPlan_t {
    *   Sort order (SORT_DIR_ASC or SORT_DIR_DESC)
    *
    */
-  inline void ExecSort(OutputTensor &a_out,
-                       const InputOperator &a,
-                       const cudaStream_t stream,
-                       const SortDirection_t dir = SORT_DIR_ASC)
-  {
+ inline void OptimizedExecSort(
```

Comment:
Can we just replace ExecSort if this is faster all-around?

Reply:
Optimized ExectSort is only faster in certain cases ( comment in ExectSort explaining cases)
`  // use optimized segmented sort if:
  //    - it is available (cub > 1.4)
  //    - it is greater than Rank 1 Tensor
  //    - it is < 8192 in all dims`
  
  Also OptimizedExecSort requires cub 1.4 or > which translates to CUDA 11.6 I believe? which I think is higher than our current CUDA requirement. 
  
  we could remove the ifdefs if we're ok moving the CUDA requirement up, but we'll still only want to use the Segmented calls for the conditions above.
  
  is the name misleading? I can rename to something like `ExecSegmentSort`